actors/test/miner_scenario_test.go

```diff
@@ -0,0 +1,315 @@
+	}
+	_, code = v.ApplyMessage(addrs[0], minerAddrs.RobustAddress, big.Zero(), builtin.MethodsMiner.PreCommitSector, &preCommitParams)
+	require.Equal(t, exitcode.Ok, code)
+
+	// assert successful precommit invocation
+	vm.ExpectInvocation{
+		// Original send to storage power actor
+		To:     minerAddrs.IDAddress,
+		Method: builtin.MethodsMiner.PreCommitSector,
+		Params: vm.ExpectObject(&preCommitParams),
+		SubInvocations: []vm.ExpectInvocation{
+			{To: builtin.RewardActorAddr, Method: builtin.MethodsReward.ThisEpochReward},
+			{To: builtin.StoragePowerActorAddr, Method: builtin.MethodsPower.CurrentTotalPower},
+			{To: builtin.StorageMarketActorAddr, Method: builtin.MethodsMarket.VerifyDealsForActivation}},
+	}.Matches(t, v.Invocations()[0])
+
+	// find information about precommited sector
+	var minerState miner.State
+	err = v.GetState(minerAddrs.IDAddress, &minerState)
+	require.NoError(t, err)
+
+	precommit, found, err := minerState.GetPrecommittedSector(v.Store(), sectorNumber)
+	require.NoError(t, err)
+	require.True(t, found)
+
+	// advance time to max seal duration
+	proveTime := v.GetEpoch() + miner.MaxSealDuration[sealProof]
+	v, dlInfo := vm.AdvanceTillEpoch(t, v, minerAddrs.IDAddress, proveTime)
+
+	//
+	// overdue precommit
+	//
+
+	t.Run("missed prove commit results in precommit expiry", func(t *testing.T) {
+		// advanced one more deadline so precommit is late
+		tv, _ := v.WithEpoch(dlInfo.Close)
+		require.NoError(t, err)
+
+		// run cron which should expire precommit
+		_, code = tv.ApplyMessage(builtin.SystemActorAddr, builtin.CronActorAddr, big.Zero(), builtin.MethodsCron.EpochTick, nil)
+		require.Equal(t, exitcode.Ok, code)
+
+		vm.ExpectInvocation{
+			// Original send to storage power actor
+			To:     builtin.CronActorAddr,
+			Method: builtin.MethodsCron.EpochTick,
+			SubInvocations: []vm.ExpectInvocation{
+				{To: builtin.StoragePowerActorAddr, Method: builtin.MethodsPower.OnEpochTickEnd, SubInvocations: []vm.ExpectInvocation{
+					{To: minerAddrs.IDAddress, Method: builtin.MethodsMiner.OnDeferredCronEvent, SubInvocations: []vm.ExpectInvocation{
+						{To: builtin.RewardActorAddr, Method: builtin.MethodsReward.ThisEpochReward},
+						{To: builtin.StoragePowerActorAddr, Method: builtin.MethodsPower.CurrentTotalPower},
+
+						// The call to burnt funds indicates the overdue precommit has been penalized
+						{To: builtin.BurntFundsActorAddr, Method: builtin.MethodSend, Value: vm.ExpectAttoFil(precommit.PreCommitDeposit)},
+						{To: builtin.StoragePowerActorAddr, Method: builtin.MethodsPower.EnrollCronEvent},
+					}},
+					//{To: minerAddrs.IDAddress, Method: builtin.MethodsMiner.ConfirmSectorProofsValid},
+					{To: builtin.RewardActorAddr, Method: builtin.MethodsReward.UpdateNetworkKPI},
+				}},
+				{To: builtin.StorageMarketActorAddr, Method: builtin.MethodsMarket.CronTick},
+			},
+		}.Matches(t, tv.Invocations()[0])
```

Comment:
I would like to see some high-level state checks, e.g. that the miner's IP is still zero and PCD is now also 0, and power actor total pledge also zero. (We need _some_ state inspection in lieu of read-only methods). Possibly extract them behind a method (like Go-filecoin's state view) so that future changes are easy to ingest here.